source/validate_instruction.cpp

```diff
@@ -131,6 +131,68 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the number of characters in a string doesn't exceed the limit.
+spv_result_t LimitCheckString(ValidationState_t& _,
+                              const spv_parsed_instruction_t* inst) {
+  // By definition, num_words of an operand is a 16-bit unsigned which will not
+  // exceed the limit of characters for a string literal.
+  // This function checks that a NULL termination must be found before reaching
+  // the end of the string words.
+  for (int i = 0; i < inst->num_operands; ++i) {
+    auto operand = inst->operands[i];
+    if (SPV_OPERAND_TYPE_LITERAL_STRING == operand.type) {
+      bool null_terminated = false;
```

Comment:
Save yourself a loop by leveraging strlen.
There's a null byte if strlen is less than 4 * num words.
```
const size_t width_in_bytes = operand.num_words * sizeof(uint32_t);
const bool null_terminated = std::strnlen_s(reinterpret_cast<const char*>(inst->words + operand.offset)), width_in_bytes) == width_in_bytes;
```